Framework/Algorithms/test/PDFFourierTransform2Test.h

```diff
@@ -260,6 +289,132 @@ class PDFFourierTransform2Test : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(alg.determineMinIndex(0.0, X, endZeroValuesY), 3);
     TS_ASSERT_EQUALS(alg.determineMaxIndex(20.0, X, endZeroValuesY), 97);
   }
+
+  void test_PDFTypes_fwd() {
+    API::MatrixWorkspace_sptr ws = createWS(20, 0.1, "CheckResult2", "MomentumTransfer");
+
+    // create the material
+    using StringProperty = Mantid::Kernel::PropertyWithValue<std::string>;
+    using FloatProperty = Mantid::Kernel::PropertyWithValue<double>;
+
+    auto material = std::make_shared<Mantid::Kernel::PropertyManager>();
+    material->declareProperty(std::make_unique<StringProperty>("ChemicalFormula", "V"), "");
+    material->declareProperty(std::make_unique<FloatProperty>("SampleNumberDensity", 0.07192), "");
+
+    auto const rho0 = 0.07192;
+
+    // set the sample information
+    Mantid::DataHandling::SetSample setsample;
+    setsample.initialize();
+    setsample.setProperty("InputWorkspace", ws);
+    setsample.setProperty("Material", material);
+    setsample.execute();
+
+    // check g(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_gr = standard_ws_ptr(ws, "g(r)", "Forward");
+    const auto little_gofR = pdfws_gr->y(0);
+    const auto gofR_comparison = 3.2015617108;
+    TS_ASSERT_DELTA(little_gofR[10], gofR_comparison, 1e-8);
+
+    // check G(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_big_gr = standard_ws_ptr(ws, "G(r)", "Forward");
+    const auto big_gofR = pdfws_big_gr->y(0);
+    const auto R_G = pdfws_big_gr->x(0);
+    const auto calculated_big_gofR = (gofR_comparison - 1) * 4. * M_PI * rho0 * R_G[10];
+    TS_ASSERT_DELTA(big_gofR[10], calculated_big_gofR, 1e-8);
+
+    // check RDF(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_rdf_r = standard_ws_ptr(ws, "RDF(r)", "Forward");
+    const auto rdfofR = pdfws_rdf_r->y(0);
+    const auto R_RDF = pdfws_rdf_r->x(0);
+    const auto calculated_rdfofR = gofR_comparison * 4. * M_PI * R_RDF[10] * rho0 * R_RDF[10];
+
+    TS_ASSERT_DELTA(rdfofR[10], calculated_rdfofR, 1e-8);
+
+    //// check G_k(r) returns correct value
+    DataObjects::Workspace2D_sptr pdfws_gkr = standard_ws_ptr(ws, "G_k(r)", "Forward");
+    const auto gkofR = pdfws_gkr->y(0);
+    const Kernel::Material &material2 = pdfws_gkr->sample().getMaterial();
+    const auto factor = 0.001 * pow(material2.cohScatterLength(), 2);
+    const auto calculated_gkofR = (gofR_comparison - 1) * factor;
+
+    TS_ASSERT_DELTA(gkofR[10], calculated_gkofR, 1e-8);
+  }
+
+  void test_PDFTypes_bkwd() {
+    API::MatrixWorkspace_sptr ws = createWS(20, 0.1, "CheckResult3", "MomentumTransfer");
+
+    // shared values for tests
+    std::vector<double> x(2, 2.0);
+    std::vector<double> dy(2, 0.0);
+    std::vector<double> dx(2, 0.0);
+    const double rho0 = 1.0;
+    const double cohScatLen = 1.0;
+    const double factor1 = 4. * M_PI * rho0;
+    const double single_x = 2.0;
+
+    // set up initial y values
+    std::vector<double> y(2, 5.0);
```

Comment:
you could use single_x here I think instead of repeating the value 2

Reply:
Just spotted your latest commit. My comment here was wrong because the 2 here is the size. Was thinking it was somehow an (x,y) value pair which it isn't